tests/rptest/services/kgo_verifier_services.py

```diff
@@ -53,91 +52,236 @@ def __init__(self, context, redpanda, topic, msg_size, custom_node):
+        self._status_thread = None
+
+        # Permit the subprocess to exit, and wait for it to do so
+        self.logger.debug(f"wait_node {self.who_am_i()}: requesting shutdown")
+        r = requests.get(self._remote_url(node, "shutdown"))
+        r.raise_for_status()
+        self.logger.debug(
+            f"wait_node {self.who_am_i()}: waiting for process to terminate")
+        wait_until(lambda: not node.account.exists(f"/proc/{self._pid}"),
+                   timeout_sec=10,
+                   backoff_sec=0.5)
+
+        return True
+
+    def _remote_url(self, node, path):
+        assert self._remote_port is not None
+        return f"http://{node.account.hostname}:{self._remote_port}/{path}"
 
     def allocate_nodes(self):
         if self.use_custom_node:
             return
         else:
-            return super(KgoVerifierServiceService, self).allocate_nodes()
+            return super(KgoVerifierService, self).allocate_nodes()
 
     def free(self):
         if self.use_custom_node:
             return
         else:
-            return super(KgoVerifierServiceService, self).free()
+            return super(KgoVerifierService, self).free()
 
 
-class ConsumerStatus:
-    def __init__(self, valid_reads, invalid_reads, out_of_scope_invalid_reads):
+class StatusThread(threading.Thread):
+    INTERVAL = 5
+
+    def __init__(self, parent: Service, node, status_cls, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._parent = parent
+        self._node = node
+        self._status_cls = status_cls
+        self._last_read = threading.Event()
+        self._stop_now = threading.Event()
+        self._ex = None
+        self._ready = False
+
+    @property
+    def errored(self):
+        return self._ex is not None
+
+    @property
+    def who_am_i(self):
+        return self._parent.who_am_i()
+
+    @property
+    def logger(self):
+        return self._parent.logger
+
+    def raise_on_error(self):
+        if self._ex is not None:
+            raise self._ex
+
+    def run(self):
+        try:
+            # Don't assume the HTTP port will be open instantly on startup (although it should be open very soon)
+            self.await_ready()
+
+            self.poll_status()
+        except Exception as ex:
+            self._ex = ex
+            self.logger.exception(
+                f"Error reading status from {self.who_am_i} on {self._node.name}"
+            )
+
+    def is_ready(self):
+        try:
+            r = requests.get(self._parent._remote_url(self._node, "status"))
+        except (requests.exceptions.RequestException,
+                requests.exceptions.ConnectionError) as e:
+            self.logger.debug(
+                f"Status endpoint {self.who_am_i} not ready: {e}")
+            return False
+        except Exception as e:
+            self.logger.debug(
+                f"Status endpoint {self.who_am_i} not ready: {e}")
+            return False
```

Comment:
I might be missing something -- any reason the RequestException and ConnectionError cases are split out if the handling is seemingly the same?

Reply:
This was me debugging a case where I didn't seem to be getting a clean `requests` exception.  I've changed this to just do the `Exception` path, because I don't trust requests to catch all socket-level exceptions